pkg/apis/certmanager/v1beta1/types_certificate.go

```diff
@@ -206,6 +179,34 @@ type CertificatePrivateKey struct {
 	// Default is 'Never' for backward compatibility.
 	// +optional
 	RotationPolicy PrivateKeyRotationPolicy `json:"rotationPolicy,omitempty"`
+
+	// The private key cryptography standards (PKCS) encoding for this
+	// certificate's private key to be encoded in.
+	// If provided, allowed values are "pkcs1" and "pkcs8" standing for PKCS#1
+	// and PKCS#8, respectively.
+	// Defaults to PKCS#1 if not specified.
+	Encoding PrivateKeyEncoding `json:"encoding,omitempty"`
+
+	// Algorithm is the private key algorithm of the corresponding private key
+	// for this certificate. If provided, allowed values are either "rsa" or "ecdsa"
+	// If `algorithm` is specified and `size` is not provided,
+	// key size of 256 will be used for "ecdsa" key algorithm and
+	// key size of 2048 will be used for "rsa" key algorithm.
+	// +optional
+	Algorithm PrivateKeyAlgorithm `json:"algorithm,omitempty"`
```

Comment:
Ditto. Shouldn't this be a pointer, since it's optional?

Reply:
Let's consolidate discussion here https://github.com/jetstack/cert-manager/pull/3038#discussion_r448965803 😄